### PR TITLE
feat: complete draft model tuning

### PR DESCRIPTION
**Pull request by nicovandenhooff**
Add draft analysis for
- 8: non-linear models
- 10: hyperparameter opt
- 11: interpretation and feat importance
- 12: test set results

Note: I've left in some `CountVectorizer` code but have it commented out now due to dimensionality problems.  I've also commented out `RandomForest` in the hyperparameter optimization, we may want to replace this with a faster non-linear model, takes extremely long to train.

### CODE REVIEW

**Review by mel-liow — APPROVED** [review]
Coool looks good!
